app/src/lib/stores/helpers/tutorial-assessor.ts

```diff
@@ -0,0 +1,145 @@
+import { IRepositoryState } from '../../app-state'
+import { TutorialStep } from '../../../models/tutorial-step'
+import { TipState } from '../../../models/tip'
+import { ExternalEditor } from '../../editors'
+import { setBoolean, getBoolean } from '../../local-storage'
+
+const skipInstallEditorKey = 'tutorial-install-editor-skipped'
+const skipCreatePullRequestKey = 'tutorial-skip-create-pull-request'
+
+/**
+ * Used to determine which step of the onboarding
+ * tutorial the user needs to complete next
+ *
+ * Stores some state that only it needs to know about. The
+ * actual step result is stored in App Store so the rest of
+ * the app can access it.
+ */
+export class OnboardingTutorialAssessor {
+  /** Has the user opted to skip the install editor step? */
+  private installEditorSkipped: boolean = getBoolean(
+    skipInstallEditorKey,
+    false
+  )
+  /** Has the user opted to skip the create pull request step? */
+  private createPRSkipped: boolean = getBoolean(skipCreatePullRequestKey, false)
+
+  public constructor(
+    /** Method to call when we need to re-check for an editor */
+    private resolveCurrentEditor: () => Promise<void>,
+    /** Method to call when we need to get the current editor */
+    private getResolvedExternalEditor: () => ExternalEditor | null
+  ) {}
+
+  /** Determines what step the user needs to complete next in the Onboarding Tutorial */
+  public async getCurrentStep(
+    isTutorialRepo: boolean,
+    repositoryState: IRepositoryState
+  ): Promise<TutorialStep> {
+    if (!isTutorialRepo) {
+      return TutorialStep.NotApplicable
+    } else if (!(await this.isEditorInstalled())) {
+      return TutorialStep.PickEditor
+    } else if (!this.isBranchCheckedOut(repositoryState)) {
+      return TutorialStep.CreateBranch
+    } else if (!this.hasChangedFile(repositoryState)) {
+      return TutorialStep.EditFile
+    } else if (!this.hasMultipleCommits(repositoryState)) {
+      return TutorialStep.MakeCommit
+    } else if (!this.commitPushed(repositoryState)) {
+      return TutorialStep.PushBranch
+    } else if (!this.pullRequestCreated(repositoryState)) {
+      return TutorialStep.OpenPullRequest
+    } else {
+      return TutorialStep.AllDone
+    }
+  }
+
+  private async isEditorInstalled(): Promise<boolean> {
+    if (this.installEditorSkipped || this.getResolvedExternalEditor()) {
+      return true
+    } else {
+      await this.resolveCurrentEditor()
+      return !!this.getResolvedExternalEditor()
```

Comment:
Can we be explicit about what we're checking for here instead of relying on the potentially ambiguous truthiness.

```suggestion
      return this.getResolvedExternalEditor() !== null
```